### PR TITLE
Speech stream handling in a statechart

### PR DESCRIPTION
**Pull request by vladmaraev**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -490,6 +613,9 @@ function App({ domElement }: any) {
             confidence: context.recResult[0]["confidence"],
           });
         },
+        addFiller: (context: SDSContext) => {
```
Comment (vladmaraev) [inline src/index.tsx:616]: it is better to implement it with `assign`

**Review by vladmaraev — COMMENTED** [review]
I am a bit unsure how we distinguish the situation of empty buffer when there are no new tokens ("Word." -> agenda = "Word.", buffer is empty) from the situation when there is "[DONE]" token. 

And see other comments :)

File: src/index.tsx [hunk 2]
```diff
@@ -142,17 +143,108 @@ const machine = Machine<SDSContext, any, SDSEvent>({
...
+                                console.debug("speakingIdle", context.buffer, event),
+                        ],
+                        always: [
+                            {
+                                target: "speaking",
+                                cond: "punctuationInBuffer"
+                            },
+                        ],
+                        after: {
+                            500: {
```
[the remark this answers is not in the document]
Comment (fredrik-talkamatic) [reply src/index.tsx]: Should of course be configurable. Will write task for it.

File: src/index.tsx [hunk 3]
```diff
@@ -389,10 +482,26 @@ function App({ domElement }: any) {
...
+        chunkIsNotDone: (_context, event) => {
+            return event.value !== "[DONE]"
+        },
+        bufferIsEmpty: (context, event) => {
+            if (context.buffer === "") {
+                console.debug("empty!");
+            } else {
+                console.debug("nonempty!", context.buffer);
+            }
+            return context.buffer === ""
```
[the remark this answers is not in the document]
Comment (fredrik-talkamatic) [reply src/index.tsx]: It works, so leaving that for a later beautification change :)

**Pull request merged**